src/include/channel.h

```diff
@@ -276,7 +282,16 @@ class ShmChannel {
     int retries = 5;
     int ret;
     do {
-      ret = __machnet_channel_buf_free_bulk(ctx_, 1, index);
+      // clear and free buffer
+      MachnetMsgBuf_t *msg_buf = __machnet_channel_buf(ctx_, index[0]);
+      __machnet_channel_buf_init(msg_buf);
+      if (cached_buf_indices.size() < CACHED_BUF_SIZE) {
+        cached_buf_indices.push_back(index[0]);
+        cached_bufs.push_back(msg_buf);
+        ret = 1;
```

Comment:
Moved outside `do {} while ()` and just `return true` since the function's supposed to return a bool type.

See commit 8e21fc50605f63db8ccca14c5eec2420e14c129d.